haystack_experimental/components/generators/anthropic/chat/chat_generator.py

```diff
@@ -0,0 +1,468 @@
+    format in input and output.
+
+    You can customize how the text is generated by passing parameters to the
+    Anthropic API. Use the `**generation_kwargs` argument when you initialize
+    the component or when you run it. Any parameter that works with
+    `anthropic.Message.create` will work here too.
+
+    For details on Anthropic API parameters, see
+    [Anthropic documentation](https://docs.anthropic.com/en/api/messages).
+
+    Usage example:
+    ```python
+    from haystack_experimental.components.generators.anthropic import AnthropicChatGenerator
+    from haystack_experimental.dataclasses import ChatMessage
+
+    generator = AnthropicChatGenerator(model="claude-3-5-sonnet-20240620",
+                                       generation_kwargs={
+                                           "max_tokens": 1000,
+                                           "temperature": 0.7,
+                                       })
+
+    messages = [ChatMessage.from_system("You are a helpful, respectful and honest assistant"),
+                ChatMessage.from_user("What's Natural Language Processing?")]
+    print(generator.run(messages=messages))
+    ```
+    """
+
+    def __init__(
+        self,
+        api_key: Secret = Secret.from_env_var("ANTHROPIC_API_KEY"),
+        model: str = "claude-3-5-sonnet-20240620",
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        ignore_tools_thinking_messages: bool = True,
+        tools: Optional[List[Tool]] = None,
+    ):
+        """
+        Creates an instance of AnthropicChatGenerator.
+
+        :param api_key: The Anthropic API key. You can set it with an environment variable `ANTHROPIC_API_KEY`,
+            or pass it directly during initialization.
+        :param model: The name of the Anthropic model to use. Defaults to "claude-3-opus-20240229".
+        :param streaming_callback: A callback function that is called when a new token is received from the stream.
+            The callback function accepts StreamingChunk as an argument.
+        :param generation_kwargs: Additional parameters to use for the model. These parameters are sent directly to
+            the Anthropic API. See Anthropic's documentation for more details on available parameters.
+             Supported generation_kwargs parameters are:
+            - `system`: The system message to be passed to the model.
+            - `max_tokens`: The maximum number of tokens to generate.
+            - `metadata`: A dictionary of metadata to be passed to the model.
+            - `stop_sequences`: A list of strings that the model should stop generating at.
+            - `temperature`: The temperature to use for sampling.
+            - `top_p`: The top_p value to use for nucleus sampling.
+            - `top_k`: The top_k value to use for top-k sampling.
```

Comment:
This part is not copied from https://github.com/deepset-ai/haystack-core-integrations/blob/7eb062c63dcb6dfa6bbaa4ba393e76ff39715ac4/integrations/anthropic/src/haystack_integrations/components/generators/anthropic/chat/chat_generator.py#L89
and there are some errors.

Reply:
Yeah, I copied most but not extra_headers. As that's related to prompt cache-ing. Do you think we should have 1:1 match between the two versions here?

In which sense errors?

Reply:
```
        :param api_key: The Anthropic API key. You can set it with an environment variable `ANTHROPIC_API_KEY`,
            or pass it directly during initialization.
        :param model: The name of the Anthropic model to use. Defaults to "claude-3-opus-20240229".
```

- API key cannot be passed directly (it must be wrapped in a `Secret`).
- We usually do not include defaults in docstring. In any case, the default model referenced in the docstring is not the same as in the code.